staging/src/k8s.io/mount-utils/mount_linux.go

```diff
@@ -637,6 +638,46 @@ func SearchMountPoints(hostSource, mountInfoPath string) ([]string, error) {
 	return refs, nil
 }
 
+// IsMountPoint determines if a directory is a mountpoint.
+// It should return ErrNotExist when the directory does not exist.
+// IsMountPoint is more expensive than IsLikelyNotMountPoint.
+// IsMountPoint detects bind mounts in linux.
+// IsMountPoint enumerates all the mountpoints using List() and
+// the list of mountpoints may be large, then it uses
+// isMountPointMatch to evaluate whether the directory is a mountpoint.
+func (mounter *Mounter) IsMountPoint(file string) (bool, error) {
+	isMnt, sure, isMntErr := mountinfo.MountedFast(file)
+	if sure == true && isMntErr == nil {
+		return isMnt, nil
+	}
+	if isMntErr != nil && os.IsPermission(isMntErr) {
+		// We were not allowed to do the simple stat() check, e.g. on NFS with
+		// root_squash. Fall back to /proc/mounts check below.
+		return false, nil
```

Comment:
The comment and the code does not match. There is no fall back to /proc/mounts here.